drivers/usb-common.h

```diff
@@ -170,6 +170,11 @@
  #define ERROR_OVERFLOW		LIBUSB_ERROR_OVERFLOW
  #define ERROR_PIPE			LIBUSB_ERROR_PIPE
  #define ERROR_TIMEOUT		LIBUSB_ERROR_TIMEOUT
+ #define ERROR_NO_MEM   LIBUSB_ERROR_NO_MEM
+ #define ERROR_INVALID_PARAM  LIBUSB_ERROR_INVALID_PARAM
+ #define ERROR_INTERRUPTED    LIBUSB_ERROR_INTERRUPTED
+ #define ERROR_NOT_SUPPORTED  LIBUSB_ERROR_NOT_SUPPORTED
+ #define ERROR_OTHER          LIBUSB_ERROR_OTHER
 
```

Comment:
I am sure I've seen at least some such defines added recently, maybe not all of these though, and maybe did not get into master yet...

Would you have a chance to define equivalents for libusb-0.1 codepath (negative ints based on errno macros like `-EINVAL` etc.) and/or fence the use of new values via `ifdef WITH_LIBUSB_1_0` if there are no equivalents in older API?

Reply:
You are right and probably think to my recent commit [d5f3e17](https://github.com/networkupstools/nut/pull/1422/commits/d5f3e17b76fff55959ce4d9b3d81f3f6e39d8366) relative to issue [#1422]( https://github.com/networkupstools/nut/pull/1422).

In this commit I have already defined all known error codes for libusb-0.1 (relative to https://github.com/libusb/libusb-compat-0.1/blob/master/libusb/core.c) and libusb-1.0 (relative to error codes defined in https://libusb.sourceforge.io/api-1.0/group__libusb__misc.html)

Later, while investigating **Eaton MGE Ellipse Max 1500** unexpected **load off**, I note that libusb errors are also defined in [drivers/usb-common.h](https://github.com/networkupstools/nut/blob/master/drivers/usb-common.h) and complete them trying to define these errors exhaustively for  further developpers.

In issue [#1422](https://github.com/networkupstools/nut/pull/1422), I only changed insufficient memory error handling to try to reconnect. I also added an upsdebug message returning error code corresponding to libusb error code as default.

From my code analysis , to not destabilize the code, I think it's enough for the moment.